File: aa25.2/lista5/j.py
import math
testes = int(input())

tamanho = 10000000
crivo = [False] * tamanho
linha = 2
coluna = 1
piram = {2 : (1, 1)}

crivo[2::2] = [True] * len(crivo[2::2])

for i in range(3, int(1 + tamanho**0.5), 2):
    if not crivo[i]:
        crivo[i * i:: 2 * i] = [True] * len(crivo[i * i:: 2* i])
        
for i in range(3, tamanho, 2):
    if not crivo[i]:
        piram[i] = (linha, coluna)

        if coluna == linha:
            linha += 1
            coluna = 1

        else:
            coluna += 1

for i in range(testes):
    numero = int(input())
    
    if crivo[i]:
        print(-1)
    else:
        print(piram[numero][0], piram[numero][1])
        

File: lista5/j.cpp
#include <bits/stdc++.h>

#define ll long long

using namespace std;


void solve(){
    ll tamanho = 1e7 + 1;
    ll numero;
    cin >> numero;
    vector<bool> crivo(tamanho);
    for(int i = 2; i * i <= tamanho; i++){
        if(!crivo[i]){
            for(int j = i * i; j < tamanho; j += i){
                crivo[j] = true;
            }
        }
    }
    
    int linha = 2;
    int coluna = 1;
    int limite = 2;
    map<ll, pair<int, int>> elementos;
    elementos[2] = {1, 1};
    for(int i = 3; i <= numero; i += 2){
        if(!crivo[i]){
            elementos[i] = {linha, coluna};
            if(coluna == limite){
                coluna = 1;
                linha ++;
                limite ++;
            }
            else{
                coluna ++;
            }
        }

    }
    if(crivo[numero]){
        cout << -1;
    }
    else{
        cout << elementos[numero].first << " " << elementos[numero].second;
    }

}

int main(){
    int testes;
    cin >> testes;
    while(testes--){
        solve();
    }

    
    return 0;
}